pkg/cmd/target.go

```diff
@@ -628,8 +641,24 @@ func getKubeConfigOfClusterType(clusterType TargetKind) (pathToKubeconfig string
 
 // getKubeConfigOfCurrentTarget returns the path to the kubeconfig of current target
 func getKubeConfigOfCurrentTarget() (pathToKubeconfig string) {
+	var targetReal Target
 	var target Target
-	ReadTarget(pathTarget, &target)
+	ReadTarget(pathTarget, &targetReal)
+
+	if len(targetReal.Target) == 1 && targetReal.Stack()[0].Kind == "namespace" {
+		panic("the target has only namespace, this is invalid, at least one garden needs to be targeted before using namespace")
```

Comment:
Can you replace `panic` function with `return errors.New("error message")`? User will be confused by the stack trace.